client/src/pages/recording-view/components/plugins-pane.tsx

```diff
@@ -96,15 +102,39 @@ export const PluginsPane = () => {
       custom_params: {},
     };
 
-    if (useCloudStorage && connectionInfo) {
+    
+    if(useCloudStorage && publicDataSource) {
+      body = {
+        samples_b64: [],
+        samples_cloud: [
+          {
+            account_name: account,
+            container_name: container,
+            file_path: meta.getFileName(),
+            sas_token: "",
+            sample_rate: sampleRate,
+            center_freq: freq,
+            data_type: MimeTypes[meta.getDataType()],
+            byte_offset: byte_offset,
+            byte_length: byte_length,
+          },
+        ],
+        custom_params: {
+          start_freq: annotation ? annotation['core:freq_lower_edge'] : null,
+          end_freq: annotation ? annotation['core:freq_upper_edge'] : null,
+        },
+      };
+    }
+
+    if (useCloudStorage && dataSource) {
```

Comment:
I would just do:

if (useCloudStorage) 

once.

change sas_token to be null for now, then when the implementation for 'get_blob_token' is done, tie that in